File: java/sdk/src/main/java/cn/huidu/device/sdk/data/program/other/Border.java
package cn.huidu.device.api.sdk.data.program.other;

public class Border {
    private int type;
    private int speed;
    private Effect effect;

    public Border() {
        type = 0;
        speed = 5;
        effect = Effect.Static;
    }

    public Border(int type) {
        this();
        this.type = type;
    }

    public Border(int type, int speed) {
        this(type);
        this.speed = speed;
    }

    public Border(int type, int speed, Effect effect) {
        this(type, speed);
        this.effect = effect;
    }

    public Border(Border other) {
        this(other.type, other.speed, other.effect);
    }

    public enum Effect {
        Rotate("rotate"),/// < 旋转
        Blink("blink"),  /// < 闪烁
        Static("null");  /// < 静止

        private final String value;

        public String getValue() {
            return value;
        }

        Effect(String value) {
            this.value = value;
        }
    }

    public int getType() {
        return type;
    }

    public void setType(int type) {
        this.type = type;
    }

    public int getSpeed() {
        return speed;
    }

    public void setSpeed(int speed) {
        this.speed = speed;
    }

    public Effect getEffect() {
        return effect;
    }

    public void setEffect(Effect effect) {
        this.effect = effect;
    }
}

File: python/sdk/data/other/Border.py
from enum import Enum


class Border:
    class Speed(Enum):
        Slow = "slow"
        Middle = "middle"
        Fast = "fast"

        def getValue(self):
            return self.value

    class Effect(Enum):
        Rotate = "rotate"
        Blink = "blink"
        Static = "null"

        def getValue(self):
            return self.value

    def __init__(self, *args):
        if len(args) == 1 and isinstance(args[0], Border):
            other = args[0]
            self.type = other.type
            self.speed = other.speed
            self.effect = other.effect
        else:
            self.type = args[0] if len(args) >= 1 else 0
            self.speed = args[1] if len(args) >= 2 else self.Speed.Middle
            self.effect = args[2] if len(args) >= 3 else self.Effect.Static

    def getType(self):
        return self.type

    def setType(self, type_val):
        self.type = type_val

    def getSpeed(self):
        return self.speed

    def setSpeed(self, speed):
        self.speed = speed

    def getEffect(self):
        return self.effect

    def setEffect(self, effect):
        self.effect = effect